app/react/kubernetes/queries/useEvents.ts

```diff
@@ -5,20 +5,25 @@ import { EnvironmentId } from '@/react/portainer/environments/types';
 import axios from '@/portainer/services/axios';
 import { withError } from '@/react-tools/react-query';
 
-import { parseKubernetesAxiosError } from '../../axiosError';
+import { parseKubernetesAxiosError } from '../axiosError';
 
-async function getNamespaceEvents(
+type Params = {
+  labelSelector?: string;
+  fieldSelector?: string;
+};
+
+async function getEvents(
   environmentId: EnvironmentId,
-  namespace: string,
-  labelSelector?: string
+  namespace?: string,
+  params?: Params
```

Comment:
I think you can put namespace into params too if it's optional. 
this way you prevent calling it like `getEvents(envId, undefined, {something})`

Reply:
Sure, updated